engine/gamesys/src/gamesys/components/comp_particlefx.cpp

```diff
@@ -285,43 +290,43 @@ namespace dmGameSystem
         dmParticle::HParticleContext particle_context = pfx_world->m_ParticleContext;
 
         dmRender::HMaterial material = GetRenderMaterial(render_context, first);
-
         dmGraphics::HVertexDeclaration vx_decl = dmRender::GetVertexDeclaration(material);
-        uint32_t vx_stride = dmGraphics::GetVertexDeclarationStride(vx_decl);
-        uint32_t max_gpu_size = pfx_context->m_MaxParticleCount * 6 * vx_stride;
-        uint32_t max_cpu_size = dmMath::Min(pfx_context->m_MaxParticleBufferCount, pfx_context->m_MaxParticleCount) * 6 * vx_stride;
 
-        if (pfx_world->m_VertexBufferData.Capacity() < max_cpu_size)
-        {
-            pfx_world->m_VertexBufferData.SetCapacity(max_cpu_size);
-        }
+        dmGraphics::VertexAttributeInfos emitter_attribute_info = {};
+        dmGraphics::VertexAttributeInfos material_attribute_info;
+        // Same default coordinate space as the editor
+        FillMaterialAttributeInfos(material, vx_decl, &material_attribute_info, dmGraphics::COORDINATE_SPACE_WORLD);
 
-        dmArray<uint8_t> &vertex_buffer = pfx_world->m_VertexBufferData;
-        uint8_t* vb_begin = vertex_buffer.End();
+        uint32_t vx_stride = material_attribute_info.m_VertexStride;
+        uint32_t max_gpu_count = pfx_context->m_MaxParticleCount;
+        uint32_t max_cpu_count = pfx_context->m_MaxParticleBufferCount; // How many particles will fit into the scratch buffer
+        uint32_t max_gpu_size = pfx_world->m_VertexBufferSize;
+        uint32_t max_cpu_size = dmMath::Min(max_cpu_count, max_gpu_count) * 6 * vx_stride;
```

Comment:
Maybe we should use a variable instead of the 6 here, just for clarity

Reply:
There are so many 6's right now, I'm not sure it will make a different tbh.

we _do_ need to update particles to support non-wuads though, for use with atlases. That task will have to remove these constants.